### PR TITLE
Promote ReferenceGrant to Beta

### LINKED ISSUE #1448: Promote ReferenceGrant to Beta
**What would you like to be added**:
Based on discussion in recent community meetings, we believe that ReferenceGrant has met the required [graduation criteria](https://gateway-api.sigs.k8s.io/concepts/versioning/#graduation-criteria) to graduate to beta. This will require updating the types to match what we've done for GatewayClass, Gateway, and HTTPRoute, with one exception: The storage version will need to stick to v1alpha2 in this initial release.

### PR DESCRIPTION
**Pull request by nathancoleman**
**What type of PR is this?**

/kind feature
/kind api-change

**What this PR does / why we need it**:
Graduates the `ReferenceGrant` resource to `v1beta1` and the `stable` release channel

**Which issue(s) this PR fixes**:

Fixes #1448 

**Does this PR introduce a user-facing change?**:

```release-note
Promotes ReferenceGrant to the v1beta1 API and the stable release channel
```

### CODE REVIEW

**Comment by robscott** [comment]
Thanks for working on this @nathancoleman! I'm going to hold off on reviewing this since it's still in draft state, but leave another comment here when it's ready for review.

/ok-to-test

**Comment by nathancoleman** [comment]
@robscott can you take a look when you get a moment?

File: hack/invalid-examples/v1alpha2/referencegrant/missing-from.yaml
Comment (robscott) [inline hack/invalid-examples/v1alpha2/referencegrant/missing-from.yaml:1]: All these files need to move from v1alpha2 -> v1beta1 dir

File: examples/experimental/reference-grant.yaml
Comment (robscott) [inline examples/experimental/reference-grant.yaml:4]: Needs to move to standard dir

File: examples/experimental/tls-cert-cross-namespace.yaml
Comment (robscott) [inline examples/experimental/tls-cert-cross-namespace.yaml:22]: Do we have any other limitations that prevent cross-namespace cert refs? I can't think of any, which makes me think this should also be labeled as "standard" now.

File: config/crd/experimental/gateway.networking.k8s.io_referencegrants.yaml
Comment (robscott) [inline config/crd/experimental/gateway.networking.k8s.io_referencegrants.yaml:267]: Calling this out to be sure it's noticed. This approach makes sense to me and matches what we did for other resources as they initially graduated to beta.

File: apis/v1beta1/referencegrant_types.go
Comment (robscott) [inline apis/v1beta1/referencegrant_types.go:64]: Can you update the v1alpha2 type definitions to use type aliases like we're doing for the rest of the v1beta1 resources? 

https://github.com/kubernetes-sigs/gateway-api/blob/a596211672a5aed54881862dc87c8c1cad9c7bd8/apis/v1alpha2/httproute_types.go#L50

**Review by robscott — COMMENTED** [review]
Thanks for the work on this @nathancoleman!

File: examples/experimental/reference-grant.yaml
Comment (nathancoleman) [reply examples/experimental/reference-grant.yaml:4]: I've added this to `standard`; however, it's not clear if that means it should be removed from `experimental`. There's some overlap between the two. Thoughts?

**Comment by nathancoleman** [comment]
Others feel free to review. It seems that re-requesting a review from Rob removes all other reviewers

**Comment by nathancoleman** [comment]
@robscott to you when you get a moment

**Comment by robscott** [comment]
Thanks @nathancoleman! This change LGTM, but I want to get another maintainer to sign off before merging since this represents graduating a resource to beta.

/approve
/assign @shaneutt @youngnick

**Review by shaneutt — COMMENTED** [review]
Looks like we need to update GEP-709 and mark it as `standard` as well

**Comment by nathancoleman** [comment]
@shaneutt what change are you looking for there? I don't see anything in the [GEP template](https://github.com/kubernetes-sigs/gateway-api/blob/2f7ab6846818c1632f163a1f97b28199ec36fa8f/site-src/geps/gep-696.md) or other GEPs that looks related

**Comment by shaneutt** [comment]
You'll wanna update the GEP status here to `standard`:

https://github.com/kubernetes-sigs/gateway-api/blob/2f7ab6846818c1632f163a1f97b28199ec36fa8f/site-src/geps/gep-709.md?plain=1#L4

And move it under "Standard" here:

https://github.com/kubernetes-sigs/gateway-api/blob/2f7ab6846818c1632f163a1f97b28199ec36fa8f/mkdocs.yml#L86

Also I would recommend taking one quick pass over the GEP just to make sure everything in it is accurate to the current API state, and if not update it to be in sync.

**Review by shaneutt — APPROVED** [review]
/lgtm

**Pull request merged**
